common/src/main/java/com/wynntils/gui/screens/maps/GuildMapScreen.java

```diff
@@ -0,0 +1,287 @@
+import com.wynntils.mc.utils.McUtils;
+import com.wynntils.utils.BoundingBox;
+import com.wynntils.utils.KeyboardUtils;
+import com.wynntils.wynn.model.map.poi.Poi;
+import com.wynntils.wynn.model.map.poi.TerritoryPoi;
+import com.wynntils.wynn.model.territory.GuildTerritoryModel;
+import com.wynntils.wynn.model.territory.objects.GuildResource;
+import com.wynntils.wynn.model.territory.objects.GuildTerritoryInfo;
+import com.wynntils.wynn.model.territory.objects.TerritoryStorage;
+import java.util.ArrayList;
+import java.util.List;
+import net.minecraft.ChatFormatting;
+import org.lwjgl.glfw.GLFW;
+
+public class GuildMapScreen extends AbstractMapScreen {
+    private boolean resourceMode = false;
+
+    @Override
+    public void render(PoseStack poseStack, int mouseX, int mouseY, float partialTick) {
+        if (holdingMapKey
+                && !GuildMapFeature.INSTANCE.openGuildMapKeybind.getKeyMapping().isDown()) {
+            this.onClose();
+            return;
+        }
+
+        updateMapCenterIfDragging(mouseX, mouseY);
+
+        RenderSystem.setShaderColor(1f, 1f, 1f, 1f);
+
+        RenderSystem.enableDepthTest();
+
+        renderMap(poseStack, GuildMapFeature.INSTANCE.renderUsingLinear);
+
+        RenderUtils.enableScissor(
+                (int) (renderX + renderedBorderXOffset), (int) (renderY + renderedBorderYOffset), (int) mapWidth, (int)
+                        mapHeight);
+
+        renderPois(poseStack, mouseX, mouseY);
+
+        RenderSystem.disableScissor();
+
+        renderBackground(poseStack);
+
+        renderCoordinates(poseStack, mouseX, mouseY);
+
+        renderHoveredTerritoryInfo(poseStack);
+    }
+
+    @Override
+    protected void renderPois(
+            List<Poi> pois,
+            PoseStack poseStack,
+            BoundingBox textureBoundingBox,
+            float poiScale,
+            int mouseX,
+            int mouseY) {
+        hovered = null;
+
+        List<Poi> filteredPois = getRenderedPois(pois, textureBoundingBox, poiScale, mouseX, mouseY);
+
+        // Render trading routes
+        for (Poi poi : filteredPois) {
+            if (!(poi instanceof TerritoryPoi territoryPoi)) continue;
+
+            float poiRenderX = MapRenderer.getRenderX(poi, mapCenterX, centerX, currentZoom);
+            float poiRenderZ = MapRenderer.getRenderZ(poi, mapCenterZ, centerZ, currentZoom);
+
+            for (String tradingRoute : territoryPoi.getTerritoryInfo().getTradingRoutes()) {
+                TerritoryPoi routePoi =
+                        GuildTerritoryModel.getGuildTerritoryMap().get(tradingRoute);
+                if (routePoi != null) {
+                    float x = MapRenderer.getRenderX(routePoi, mapCenterX, centerX, currentZoom);
+                    float z = MapRenderer.getRenderZ(routePoi, mapCenterZ, centerZ, currentZoom);
+
+                    RenderUtils.drawLine(poseStack, CommonColors.DARK_GRAY, poiRenderX, poiRenderZ, x, z, 0, 1);
+                }
+            }
```

Comment:
Does this double render?

Reply:
Yeah but it is faster to double render than to do any calculations optimisations.

Reply:
note this explicitly

Reply:
Done